cookbook/core/intermediate/use_secrets.py

```diff
@@ -0,0 +1,112 @@
+"""
+Use secrets in a task
+----------------------
+
+This example explains how a secret can be accessed in a Flyte Task. Flyte provides different types of Secrets, as part of
+SecurityContext. But, for users writing python tasks, you can only access ``secure secrets`` either as environment variable
+or injected into a file.
+
+"""
+import os
+import flytekit
+
+# %%
+# Flytekit exposes a type/class called Secrets. It can be imported as follows.
+from flytekit import Secret, task, workflow
+
+# %%
+# Secrets consists of a name and an enum that indicates how the secrets will be accessed. If the mounting_requirement is
+# not specified then the secret will be injected as an environment variable is possible. Ideally, you need not worry
+# about the mounting requirement, just specify the ``Secret.name`` that matches the declared ``secret`` in Flyte backend
+#
+# Let us declare a secret named user_secret in a secret group ``user-info``. A secret group can have multiple secret
+# associated with the group. Optionally it may also have a group_version. The version helps in rotating secrets. If not
+# specified the task will always retrieve the latest version. Though not recommended some users may want the task
+# version to be bound to a secret version.
+
+SECRET_NAME = "user_secret"
+SECRET_GROUP = "user-info"
+
+
+# %%
+# Now declare the secret in the requests. The secret can be accessed using the :py:class:`flytekit.ExecutionParameters`,
+# through the global flytekit context as shown in the body of the method
+#
+# .. note::
+#
+#   - In case of failure to access the secret (it is not found at execution time) an error is raised.
+#   - Secrets group and key are required parameter during declaration and usage. Failure to specify will cause an
```

Comment:
parameter --> parameters
will cause an exception --> will cause the following exception